### PR TITLE
Add toml library

### PR DESCRIPTION
**Pull request by sestegra**
Implement #259 to support containerd configurations embedded into yaml files.

### CODE REVIEW

**Comment by pivotaljohn** [comment]
Hey @sestegra. Thanks for the contribution!

Looking it over, I don't see any tests.

Would you mind tossing one in there? (ala https://github.com/vmware-tanzu/carvel-ytt/blob/develop/pkg/yamltemplate/filetests/ytt-library/json.yml)

**Comment by sestegra** [comment]
Hey @pivotaljohn
I will add a similar test.

**Comment by cppforlife** [comment]
@sestegra 🔥 🔥 🔥 looking forward to this getting merged!

File: pkg/yttlibrary/toml.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	"github.com/BurntSushi/toml"
+	"github.com/k14s/starlark-go/starlark"
+	"github.com/k14s/starlark-go/starlarkstruct"
+	"github.com/k14s/ytt/pkg/orderedmap"
+	"github.com/k14s/ytt/pkg/template/core"
+	"github.com/k14s/ytt/pkg/yamlmeta"
+)
+
+var (
+	TOMLAPI = starlark.StringDict{
```
Comment (pivotaljohn) [inline pkg/yttlibrary/toml.go]:
```suggestion
var (
	// TOMLAPI contains the definition of the @ytt:toml module
	TOMLAPI = starlark.StringDict{
```

File: pkg/yttlibrary/toml.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+		"toml": &starlarkstruct.Module{
+			Name: "toml",
+			Members: starlark.StringDict{
+				"encode": starlark.NewBuiltin("toml.encode", core.ErrWrapper(tmùModule{}.Encode)),
+				"decode": starlark.NewBuiltin("toml.decode", core.ErrWrapper(tmùModule{}.Decode)),
+			},
+		},
+	}
+	TOMLKWARGS = map[string]struct{}{
```
Comment (pivotaljohn) [inline pkg/yttlibrary/toml.go:31]:
```suggestion
	// TOMLKWARGS names the expected keyword arguments for both toml.encode
	TOMLKWARGS = map[string]struct{}{
```

File: pkg/yttlibrary/toml.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+	err = encoder.Encode(val)
+
+	if err != nil {
+		return starlark.None, err
+	}
+
+	return starlark.String(buffer.String()), nil
+}
+
+func (b tmùModule) Decode(thread *starlark.Thread, f *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
```
Comment (pivotaljohn) [inline pkg/yttlibrary/toml.go:72]:
```suggestion
// Encode is a core.StarlarkFunc that parses the provided input from TOML format into dicts, lists, and scalars
func (b tmùModule) Decode(thread *starlark.Thread, f *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
```

**Review by pivotaljohn — COMMENTED** [review]
Suggesting godoc.

**Comment by pivotaljohn** [comment]
Associated reference docs:
PR: vmware-tanzu/carvel#273
preview: https://deploy-preview-273--carvel.netlify.app/ytt/docs/latest/lang-ref-ytt/#toml

**Review by pivotaljohn — APPROVED** [review]
On final review, looks like I borked the godoc suggestions.
However, what's there is enough to satisfy the linter. I'll fix the godoc "in post".

Thanks again, @sestegra. 🙏🏻

**Comment by pivotaljohn** [comment]
@sestegra — did you have a particular timeline you were hoping to see this generally released in?

We're considering delaying cutting a release for a couple of weeks or so... but wanted your input.

**Comment by sestegra** [comment]
@pivotaljohn No problem, I have no constraints.
In the meantime, I am using my own built binary.

**Pull request merged**